src/tabnote.ts

```diff
@@ -431,33 +449,40 @@ export class TabNote extends StemmableNote {
       ctx.clearRect(tab_x - 2, y - 3, glyph.getWidth() + 4, 6);
 
       if (glyph.code) {
-        Glyph.renderGlyph(ctx, tab_x, y, this.render_options.glyph_font_scale * this.render_options.scale, glyph.code);
+        // eslint-disable-next-line
+        Glyph.renderGlyph(ctx, tab_x, y, this.render_options.glyph_font_scale * this.render_options.scale!, glyph.code);
       } else {
         ctx.save();
-        ctx.setRawFont(this.render_options.font);
+        // eslint-disable-next-line
+        ctx.setRawFont(this.render_options.font!);
         const text = glyph.text.toString();
-        ctx.fillText(text, tab_x, y + 5 * this.render_options.scale);
+        // eslint-disable-next-line
+        ctx.fillText(text, tab_x, y + 5 * this.render_options.scale!);
         ctx.restore();
       }
     }
   }
 
   // The main rendering function for the entire note
-  draw() {
-    this.checkContext();
+  draw(): void {
+    const ctx = this.checkContext();
 
     if (!this.stave) {
       throw new Vex.RERR('NoStave', "Can't draw without a stave.");
     }
 
+    if (!this.stem) {
```

Comment:
This seems wrong -- you can have tab notes without stems. Does this affect the tests?

Reply:
No impact on tests. However I agree with you. I have changed the implementation and it also pass all the tests.